sql/analyze_test.go

```diff
@@ -147,6 +147,8 @@ func TestSimplifyExpr(t *testing.T) {
 
 		{`a IN (1, 1)`, `a IN (1)`},
 		{`a IN (2, 3, 1)`, `a IN (1, 2, 3)`},
+		{`a IN (1, true)`, `true`},
```

Comment:
Yeah, the expression isn't valid. Prior to `typeCheckTuple` sorting of the values would panic due to int and bool not being comparable. I think we need a `TypeCheckExpr` pass that is run before simplification.